src/engine/world/room.cpp

```diff
@@ -584,12 +585,39 @@ void Room::collectShaderLights(size_t depth)
   lightsBuffer->setData(bufferLights, gl::api::BufferUsage::StaticDraw);
 }
 
-void Room::createParticleMesh(const std::string& label,
-                              const glm::vec3& min,
-                              const glm::vec3& max,
-                              render::scene::MaterialManager& materialManager)
+void Room::regenerateDust(const std::shared_ptr<engine::Presenter>& presenter,
+                          const gslu::nn_shared<render::scene::Material>& dustMaterial,
+                          bool isDustEnabled,
+                          float dustDensityDivisor)
+{
+  if(!isDustEnabled)
+  {
+    dust = nullptr;
+    return;
+  }
+
+  if(const auto it = dustCache.find(dustDensityDivisor); it != dustCache.end())
+  {
+    dust = it->second;
+    return;
+  }
+  if(presenter != nullptr)
+    presenter->drawLoadingScreen(_("Generating Dust Particles..."));
+  auto dustNode = createParticleMesh(node->getName() + "/dust", verticesBBoxMin, 
+                                     verticesBBoxMax, dustMaterial, 
+                                     dustDensityDivisor);
+  dustCache.emplace(dustDensityDivisor, dustNode);
+  dust = dustNode;
+}
+
+std::shared_ptr<render::scene::Node> Room::createParticleMesh(const std::string& label,
+                                                              const glm::vec3& min,
+                                                              const glm::vec3& max,
+                                                              const gslu::nn_shared<render::scene::Material>& dustMaterial,
+                                                              float dustDensity)
 {
-  static const constexpr auto Resolution = (1_sectors / 12).cast<float>().get();
+  static const auto densityDivisor = 9.0f - (dustDensity / 12.5f);
+  static const auto resolution = (1_sectors / 12).cast<float>().get() * static_cast<float>(cbrt(densityDivisor));
```

Comment:
This is some convoluted math I wrote back then. The intention was this: the base grid is 12 subdivision per axis. The axis scale would then be `1/cbrt(densityDivisor)`, i.e. (ignoring type issues) the effective grid axis resolution would now be `cbrt(densityDivisor) / 12 * 1_sector`. To express this, a constant like `BaseGridAxisSubdivision = 12` would greatly help understanding this I guess.